python-avd/pyavd/_cv/client/studio.py

```diff
@@ -103,14 +105,11 @@ async def get_studio(
             time=TimeBounds(start=None, end=time),
         )
         client = StudioConfigServiceStub(self._channel)
-        try:
-            responses = client.get_all(request, metadata=self._metadata, timeout=timeout)
-            async for _response in responses:
-                # If we get here it means we got an entry with "removed: True" so no need to look further.
-                msg = "The studio was deleted in the workspace."
-                raise CVResourceNotFound(msg, f"Studio ID '{studio_id}, Workspace ID '{workspace_id}'")  # noqa: TRY301 TODO: Improve error handling
-        except Exception as e:
-            raise get_cv_client_exception(e, f"Studio ID '{studio_id}, Workspace ID '{workspace_id}'") or e
+        responses = client.get_all(request, metadata=self._metadata, timeout=timeout)
+        async for _response in responses:
+            # If we get here it means we got an entry with "removed: True" so no need to look further.
+            msg = "The studio was deleted in the workspace."
+            raise GRPCError(status=Status.NOT_FOUND, message=f"{msg}. Studio ID '{studio_id}, Workspace ID '{workspace_id}'")
```

Comment:
I would prefer not to raise GRPCErrors from something that is not really a GRPC error. We could still raise CVResourceNotFound, or create a more specific error type if we like.